Framework/Beamline/inc/MantidBeamline/ComponentInfo.h

```diff
@@ -121,6 +123,9 @@ class MANTID_BEAMLINE_DLL ComponentInfo {
   size_t sample() const;
   size_t root() const;
   double l1() const;
+  Eigen::Vector3d scaleFactor(const size_t componentIndex) const;
+  void setScaleFactor(const size_t componentIndex,
+                      const Eigen::Vector3d &scaleFactor);
```

Comment:
Not that I can think of currently. The only place that seems to be directly setting this parameter is in the algorithm I have already modified.